pkg/mls/store/store.go

```diff
@@ -53,6 +64,111 @@ func New(ctx context.Context, config Config) (*Store, error) {
 	return s, nil
 }
 
+func (s *Store) PublishIdentityUpdate(ctx context.Context, req *identity.PublishIdentityUpdateRequest) (*identity.PublishIdentityUpdateResponse, error) {
+	new_update := req.GetIdentityUpdate()
+	if new_update == nil {
+		return nil, errors.New("IdentityUpdate is required")
+	}
+
+	// TODO: Implement serializable isolation level once supported
+	if err := s.db.RunInTx(ctx, &sql.TxOptions{ /*Isolation: sql.LevelSerializable*/ }, func(ctx context.Context, tx bun.Tx) error {
+		inbox_log_entries := make([]*InboxLogEntry, 0)
+
+		if err := s.db.NewSelect().
+			Model(&inbox_log_entries).
+			Where("inbox_id = ?", new_update.GetInboxId()).
+			Order("sequence_id ASC").
+			For("UPDATE").
+			Scan(ctx); err != nil {
+			return err
+		}
+
+		if len(inbox_log_entries) >= 256 {
+			return errors.New("inbox log is full")
+		}
+
+		updates := make([]*associations.IdentityUpdate, 0, len(inbox_log_entries)+1)
+		for _, log := range inbox_log_entries {
+			identity_update := &associations.IdentityUpdate{}
+			if err := proto.Unmarshal(log.IdentityUpdateProto, identity_update); err != nil {
+				return err
+			}
+			updates = append(updates, identity_update)
+		}
+		_ = append(updates, new_update)
+
+		// TODO: Validate the updates, and abort transaction if failed
+
+		proto_bytes, err := proto.Marshal(new_update)
+		if err != nil {
+			return err
+		}
+
+		new_entry := InboxLogEntry{
+			InboxId:             new_update.GetInboxId(),
+			ServerTimestampNs:   nowNs(),
+			IdentityUpdateProto: proto_bytes,
+		}
+
+		_, err = s.db.NewInsert().
+			Model(&new_entry).
+			Returning("sequence_id").
+			Exec(ctx)
+		if err != nil {
+			return err
+		}
+
+		// TODO: Insert or update the address_log table using sequence_id
+
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return &identity.PublishIdentityUpdateResponse{}, nil
+}
+
+func (s *Store) GetInboxLogs(ctx context.Context, batched_req *identity.GetIdentityUpdatesRequest) (*identity.GetIdentityUpdatesResponse, error) {
+	reqs := batched_req.GetRequests()
+	resps := make([]*identity.GetIdentityUpdatesResponse_Response, len(reqs))
+
+	for i, req := range reqs {
+		inbox_log_entries := make([]*InboxLogEntry, 0)
+
+		err := s.db.NewSelect().
```

Comment:
Would be nice to get all of these in a single SQL query, but we can optimize that later.